cdap-app-fabric/src/main/java/io/cdap/cdap/internal/bootstrap/executor/AppCreator.java

```diff
@@ -84,7 +84,8 @@ public void execute(Arguments arguments) throws Exception {
 
   private boolean appExists(ApplicationId applicationId) throws Exception {
     try {
-      appLifecycleService.getAppDetail(applicationId);
+      // applicationId is always "-SNAPSHOT", call getLatestAppDetail
```

Comment:
if it's always -snapshot, it should be changed to ApplicationReference all the way where it is first created (looks like in `AppCreator.Arguments.getId()`) and passed as ApplicationReference.

Basically, anywhere we are creating ApplicationId without passing in a specific version, we should be creating ApplicationReference instead.

Reply:
updated